router/middleware/perm/perm.go

```diff
@@ -9,29 +9,106 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/go-vela/sdk-go/vela"
 	"github.com/go-vela/types"
-	"github.com/go-vela/worker/router/middleware/user"
+	"github.com/go-vela/worker/router/middleware/token"
 
 	"github.com/gin-gonic/gin"
 	"github.com/sirupsen/logrus"
 )
 
-// MustServer ensures the user is the vela server.
+// MustServer ensures the caller is the vela server.
 func MustServer() gin.HandlerFunc {
 	return func(c *gin.Context) {
-		u := user.Retrieve(c)
+		// retrieve the callers token from the request headers
+		tkn, err := token.Retrieve(c.Request)
+		if err != nil {
+			msg := fmt.Sprintf("error parsing token: %s", err)
+
+			err := c.Error(fmt.Errorf(msg))
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			c.AbortWithStatusJSON(http.StatusBadRequest, err.Error())
 
-		if strings.EqualFold(u.GetName(), "vela-server") {
 			return
 		}
 
-		msg := fmt.Sprintf("User %s is not a platform admin", u.GetName())
+		// retrieve the configured server address from the context
+		addr, ok := c.MustGet("server-address").(string)
+		if !ok {
+			msg := "error retrieving server address"
+
+			err := c.Error(fmt.Errorf(msg))
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			c.AbortWithStatusJSON(http.StatusInternalServerError, types.Error{Message: &msg})
+
+			return
+		}
 
-		err := c.Error(fmt.Errorf(msg))
+		// create a temporary client to validate the incoming request
+		vela, err := vela.NewClient(addr, "vela-worker", nil)
 		if err != nil {
-			logrus.Error(err)
+			msg := fmt.Sprintf("error creating vela client: %s", err)
+
+			err := c.Error(fmt.Errorf(msg))
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			c.AbortWithStatusJSON(http.StatusInternalServerError, types.Error{Message: &msg})
+
+			return
 		}
```

Comment:
Thoughts on changing this (and other error blocks like it) to:

```go
		if err != nil {
			retErr := fmt.Errorf("error creating vela client: %s", err)
		
			logrus.Error(retErr)
			
			c.AbortWithStatusJSON(http.StatusInternalServerError, retErr)

			return
		}
```
? A little more condensed, though I'm indifferent. Better than just logging it.

Reply:
also, should our pattern be to return the json wrapped error with `message:` or just the raw error as the string?

IE: `types.Error{Message: &msg}` vs `retErr` 

both patterns are used across multiple go-vela repos so its tough to say.

i can definitely go with `string` for simplicity